Initial RGBFusion Support

Cheerpipe opened the pull request:
Fixes #873

**This pull request proposes the following changes:**
- Support for RGBFusion Motherboards and compatible peripherals like VGA cards, RAM, and mices.

**Known issues/To do:**
- Bridge installation should be optional, not automatic.
- It applies same color to all areas. Bridge support single area color but Aurora Device need more work to support this feature.
- RGBFusionBridge should be a module of the project. 

Tested on a Aorus Master Z390. It is also reported to works in x570 motherboards.

Firs time coding something like this so any review/advice/critic will be very well received. 

**READ ME:** 

- This development don't use Gigabyte SDK so it supports newest motherboards with Intel X390 and x570. It uses calls to Gigabyte RGBFusion internal methods and, in some case, direct read-write to the SMBUS/MCU registers found with revere engineering. **Use under your own risk**.
- It is recommended to use with RGBFusion B19.0919 because latest version is slower (https://drive.google.com/file/d/1inUFhNrpah1WZdJQv07SwKYnVfJ6DakL/view)
- RGBFusion have to be launched at least one time every time you install a new version of RGBFusion in order to detect hardware, after that, you can close RGBFusion and disable autostart at boot because it is not needed. In fact, it SHOULD be NOT RUNNING to use Aurora with RGBFusion devices.
- RGBFusion should be closed in order to work.

**Optional:** 

- You can run `RGBFusionAuroraListener.exe --ngen` command to create a native image and save a bit of cpu usage.

Thanks in advance.

Cheerpipe committed:
Initial RGBFusion Support using RGBFusion Bridge

It requires RGBFusion 2 installed to work.
It automatically install the Bridge in the RGBFusion Folder.
RGBFusion should be closed in order to work.

Cheerpipe committed:
Merge remote-tracking branch 'antonpup/dev' into RGBFusion

Cheerpipe committed:
Delete debug lines and create a new clear device map update method.

Cheerpipe commented:
This is the source of RGBFusion Bridge: https://github.com/Cheerpipe/RGBFusionCli/tree/Rewrite

Cheerpipe committed:
Remove unnecessary validation

Cheerpipe committed:
Multi area support

Cheerpipe committed:
Configure keys for each zones

Note: RGBFusion will slowdown when changing peripherals (area index above 7). it is recommended to maintain these areas static.

Using new bridge version that reduce calls to RGBFusion when an area has not been changed.

Cheerpipe committed:
Merge branch 'dev' into RGBFusion

Mitonix commented:
Hi. You're welcome. ;D

**Is it working for you?**
It does not work. 

logs: [2020-06-01 17.43.08.log](https://github.com/antonpup/Aurora/files/4712135/2020-06-01.17.43.08.log)

**What motherboard are you using?**
I use the Gigabyte Z370 Aorus Gaming 5.

**How are working your leds?**
The colour of the leds does not change.

**Is there any artifact or glitch in your lights?**
Not tested.

**Changes are slow or fast? Dis you try some animated mode?**
Not tested.

**Are colors for separate areas working?**
No.

Frosthaven commented:
Same results as above on Aorus X570 Elite. No lighting changes, with the same crash on Aurora exit or this component disable.

Cheerpipe commented:
It looks like RGBFusion Bridge is not running.

Can you check if you have the following files?

In your Aurora folder:
            "RGBFusionBridge\RGBFusionAuroraListener.exe",
            "RGBFusionBridge\LedLib2.dll",
            "RGBFusionBridge\RGBFusionBridge.dll"

And in your RGBFusion Folder, usually "C:\Program Files (x86)\GIGABYTE\RGBFusion"
            RGBFusionAuroraListener.exe",
            "LedLib2.dll",
            "RGBFusionBridge.dll"

Important thing, RGBFusion must be closed before launch Aurora.

Can you also check if after launching Aurora you have a process running called "RGBFusionAuroraListener"?

Thanks in advance

Cheerpipe committed:
RGBFusion Bridge detection fix and File MD5 check

Cheerpipe commented:
Try new build pls.

Thanks in advance

skielbasa commented:
I also got that same issue and latency was at ~16ms.

It looks like RGBFusionBridge.dll was missing from my RGBFusion folder (the other files were there). I copied that .dll from the build zip into RGBFusion and Aurora started working. Can't seem to get get any zones working (I only tried the first 5). Any lighting bound to esc does affect the system, so it is talking to the MB and the latency is down to ~1ms. 

If you've got some sort of cheat sheet for what each area index goes to, I can tweak my config to see if zones are working.

I'm on a x570 Pro Wifi rev1 with both of the standard 3pin ARGB headers used.

---

Note: I grabbed this from this link https://ci.appveyor.com/project/antonpup/aurora/builds/33231558 which was posted 15 hours ago (as of writing this) in this thread https://github.com/antonpup/Aurora/issues/873

It looks like 2 hours before writing this you said you've got a new build. Not sure if I'm on your latest one or not.

Cheerpipe commented:
> My current version of Aurora is beta 0.7.2+815 and it works fine, but have been following what you've been doing and it's got me excited, so I tested your build.
> 
> After first launch with your version I also got that same issue, latency was at ~16ms, and colors wouldn't change.
> 
> It looks like RGBFusionBridge.dll was missing from my RGBFusion folder (the other files were there). I copied that .dll from the build zip into RGBFusion and Aurora started working. Can't seem to get get any zones working (I only tried the first 5). Any lighting bound to esc does affect the system, so it is talking to the MB and the latency is down to ~1ms.
> 
> If you've got some sort of cheat sheet for what each area index goes to, I can tweak my config to see if zones are working.
> 
> I'm on a x570 Pro Wifi rev1 with both of the standard 3pin ARGB headers used.
> 
> Note: I grabbed this from this link https://ci.appveyor.com/project/antonpup/aurora/builds/33231558 which was posted 15 hours ago (as of writing this) in this thread #873
> 
> It looks like 2 hours before writing this you said you've got a new build. Not sure if I'm on your latest one or not.
> 
> Edit: Added intro about Aurora version.

Laste version (https://ci.appveyor.com/project/antonpup/aurora/builds/33253988) should solve bridge installation / dll file copy problems.

I don't have any area index list for your motherboard, just for Aorus Master X390. It is up to you to set different Keys to areas and see wich is wich. If you do this, pls report back and post your findings.

For the time being, in general, this working for you? Lights change correctly? ()

skielbasa commented:
> I don't have any area index list for your motherboard, just for Aorus Master X390. It is up to you to test different areas and see which is which. If you do this, pls report back and post your findings.

Noted. It'll probably be a few days, but I'll see what I can find. These are supposed to represent whole zones right? Not individual LEDs? or is that something that could vary board to board as well?

Mitonix commented:
> 
> 
> It looks like RGBFusion Bridge is not running.
> 
> Can you check if you have the following files?
> 
> In your Aurora folder:
> "RGBFusionBridge\RGBFusionAuroraListener.exe",
> "RGBFusionBridge\LedLib2.dll",
> "RGBFusionBridge\RGBFusionBridge.dll"
> 
> And in your RGBFusion Folder, usually "C:\Program Files (x86)\GIGABYTE\RGBFusion"
> RGBFusionAuroraListener.exe",
> "LedLib2.dll",
> "RGBFusionBridge.dll"
> 
> Important thing, RGBFusion must be closed before launch Aurora.
> 
> Can you also check if after launching Aurora you have a process running called "RGBFusionAuroraListener"?
> 
> Thanks in advance

I tested the latest build, but I could hardly see any changes. Only the latency varies between 11 and 15 ms (a bit less than before). All files are installed and RGBFusionAuroraListener is running. But the colors of the mainboard do not change.

Frosthaven commented:
Gigabyte X570 Aorus Elite

Using the latest version, Aurora no longer crashes. However, RGBFusionAuroraListener.exe spits the pictured error out. This error can be seen on aurora first-launch or when re-enabling the RGB Fusion Device after disabling it.

![Qm5ZToV](https://user-images.githubusercontent.com/5412207/83515448-60454b80-a4a3-11ea-9e5f-632d1aefeab7.png)

Cheerpipe commented:
It looks like newer RGBFusion broke some things for me. I am using B19.0919.1. because it is faster and stable. Latest versions are slower and have hardware detection issues for a lot of users, however it has wider hardware supports. 

Are you using latest version from RGBFusion Website?

You can try using older RGBFusion version if you want. I will work on patch latest RGBFusion later. 

B19.0919.1: https://drive.google.com/file/d/1inUFhNrpah1WZdJQv07SwKYnVfJ6DakL/view

Mitonix commented:
With the older version B19.0919.1 it works for me. The latest version of RGBFusion does not work, as you mentioned before. The colors change very fast. But I cannot control the individual areas. 
Thanks :D

Cheerpipe commented:
> With the older version B19.0919.1 it works for me. The latest version of RGBFusion does not work, as you mentioned before. The colors change very fast. But I cannot control the individual areas.
> Thanks :D

Can you upload your Pro1.xml and ExtPro1.xml from your \Program Files (x86)\GIGABYTE\RGBFusion folder?

Thanks in advance

Mitonix commented:
> 
> 
> > With the older version B19.0919.1 it works for me. The latest version of RGBFusion does not work, as you mentioned before. The colors change very fast. But I cannot control the individual areas.
> > Thanks :D
> 
> Can you upload your Pro1.xml and ExtPro1.xml from your \Program Files (x86)\GIGABYTE\RGBFusion folder?
> 
> Thanks in advance

I cannot upload the xml files because Github does not support this format.

skielbasa commented:
Tried to get a start on mapping out the zones, but it seems like the key selection doesn't work right in the device options of Aurora. I had made it to index 20 setting them to a test key with a obvious color change to be easy to spot, but nothing was updating.

I restarted Aurora and the most of the board was following that test pattern. Resetting indexes or changing them to another key didn't have an effect on their color.

What this says to me is that per zone does seem to be working (not every light was following the test pattern) and that assigning indexes to a key in Aurora seems to only take effect when launching Aurora.

Thinking maybe it was the outdated Aurora build, I grabbed your latest one with the fixed bridge installation. That build broke pretty much everything. Aurora couldn't set colors (neither my main install or this test one) and RGBFusion itself wouldn't run, much like what happened to Mitonix. I reinstalled RGBFusion and its back to working as it did before running the new build. 

I was and am again on the latest RGBFusion. I'll try rolling back to the one you linked and see where that gets me.

diogotr7 committed:
Merge pull request #2061 from antonpup/dev

v0.8.0-rc2

Cheerpipe committed:
Fix peripherals area detection

Cheerpipe committed:
Support for RGBFusion B20.0430.1

Patched LedLib2.dll B20.0430.1
Update MD5 table.

Cheerpipe commented:
> would love to see this not being a requirement.

You are right. Aurora must be restarted to apply changes.

I updated the pull request with support for B20.0430.1. This version of RGBFusion could be very slow when using animated modes. Pls tray and report back.

I also do a small change than may be compatible with RGBFusion peripherals but without a gigabyte mainboard.

Cheerpipe commented:
> Tried to get a start on mapping out the zones, but it seems like the key selection doesn't work right in the device options of Aurora. I had made it to index 20 setting them to a test key with a obvious color change to be easy to spot, but nothing was updating.
> 
> I restarted Aurora and the most of the board was following that test pattern. Resetting indexes or changing them to another key didn't have an effect on their color.
> 
> What this says to me is that per zone does seem to be working (not every light was following the test pattern) and that assigning indexes to a key in Aurora seems to only take effect when launching Aurora.
> 
> Thinking maybe it was the outdated Aurora build, I grabbed your latest one with the fixed bridge installation. That build broke pretty much everything. Aurora couldn't set colors (neither my main install or this test one) and RGBFusion itself wouldn't run, much like what happened to Mitonix. I reinstalled RGBFusion and its back to working as it did before running the new build.
> 
> I was and am again on the latest RGBFusion. I'll try rolling back to the one you linked and see where that gets me.

@skielbasa do you have 20 area index O.o??? Can you pls upload your Pro1.xml and ExtPro1.xml from your c: \Program Files (x86)\GIGABYTE\RGBFusion folder?

Thanks in advance

skielbasa commented:
@Cheerpipe I'm going through it now. I did all that before realizing Aurora restarts are necessary, so I had no way of know what index controlled what zone. I have confirmed that animated lighting works fine as does changing the actual lighting for a key without having to restart. It looks like the index to key assignment is the only part that needs the restart.

UI Bug on the Aurora-v0.8.0+856-cqcmcgcc build: there is no index 7

![Annotation 2020-06-02 170502](https://user-images.githubusercontent.com/5097785/83570073-6fec8080-a4f3-11ea-9edb-d9fec8082990.png)

Note: I did downgrade to the version of RGBFusion you linked to, it saw all my hardware, so guess I'm sticking with that version now. On the plus side the animations seem less stepped.

Frosthaven commented:
Just tested the latest build as of this posting alongside the latest RGB Fusion v2.0

Results:

- index mapping stayed the same on the aorus x570 elite (0 for the top, 3 for both of the bottom indicators)
- animations stayed in sync with keyboard during testing (tested: blink, breathe, glitch)

skielbasa commented:
Zone indexing is going pretty well, but I hit another bug. I don't know if this is Aurora, RGBFusion Bridge, or some combination. After assigning color to a bunch of keys Aurora stopped updating the MB. 

Restarting the device and the app didn't help initially. Got it working again by disabling the device and killing RGBFusionAuroraListener and RGBFusion390SetColor in task manager, then re-enabling the device.

Cheerpipe committed:
Add option to add area index to ignore list

This command should be used to avoid RGBFusion Bridge control some areas. It is useful if you want to control your RAM RGB with another application because RGBFusion is slow handling RAM.

RAM is usually Area index 9. In some cases it could be 8.

Values should be comma separated.

Default ignore list is "0,8,9" (without quotes)

Cheerpipe commented:
Your Pro1.xml seems pretty normal but ExtPro1 for your GPU is crazy. Dont know if RGBFusion can handle so many areas in a smooth way.

How is your light framerate?

RAM and DLED strip as one solid area is an RGBFusion limitation by design. It does not expose any way of control separate LEDs. I managed to get some lower level control on my Z390 MB to control my ARGB fans but it appears to be different for each motherboard so I can’t write a generic device for this because it could be dangerous write wrong data in wrong registers :/

I just uploaded a new version with an option to add area indexes to a ignore list in case you want control come device (Like RAM) with their own software (RAM framerate with aurora is bad).

Cheerpipe commented:
> @Cheerpipe I'm going through it now. I did all that before realizing Aurora restarts are necessary, so I had no way of know what index controlled what zone. I have confirmed that animated lighting works fine as does changing the actual lighting for a key without having to restart. It looks like the index to key assignment is the only part that needs the restart.
> 
> UI Bug on the Aurora-v0.8.0+856-cqcmcgcc build: there is no index 7
> 
> ![Annotation 2020-06-02 170502](https://user-images.githubusercontent.com/5097785/83570073-6fec8080-a4f3-11ea-9edb-d9fec8082990.png)
> 
> Note: I did downgrade to the version of RGBFusion you linked to, it saw all my hardware, so guess I'm sticking with that version now. On the plus side the animations seem less stepped.

it os Ok. That means you don’t have area index 7. (me neither)

Cheerpipe committed:
Merge tag 'v0.8.0rc2' into RGBFusion

skielbasa commented:
> Your Pro1.xml seems pretty normal but ExtPro1 for your GPU is crazy. Dont know if RGBFusion can handle so many areas in a smooth way.
> How is your light framerate?

Not really sure how to measure that lol. If I had to guess, I'd say 10ish changes per second. Its fast enough that gradients are smooth unless the 2 colors are very different.

> RAM and DLED strip as one solid area is an RGBFusion limitation by design. It does not expose any way of control separate LEDs. I managed to get some lower level control on my Z390 MB to control my ARGB fans but it appears to be different for each motherboard so I can’t write a generic device for this because it could be dangerous write wrong data in wrong registers :/

I figured as much. RGBFusion is pretty garbage for customization. Just solids and their 15 different rainbow vomits.

> I just uploaded a new version with an option to add area indexes to a ignore list in case you want control come device (Like RAM) with their own software (RAM framerate with aurora is bad).

Funny you say that, RAM _is_ where I notice the color stepping the most. It's not enough to bother me though. I'll give the new build a test though.

> It is Ok. That means you don’t have area index 7. (me neither)

So those auto populate by what you've got? That would mean there are 11 more zones I couldn't find any lights attached to (beyond the ones I marked as ???? up there).

> I am thinking of adding a RGBVersion selector with the lastest and B19.0919.1 as an option because the last has a lot better framerate.

Is the latest version slowed down to work with the new RGBFusion? And is there a way to detect which version is installed, so the user doesn't need to set it?

---

Seriously, though, thank you for all the work you've put in. I was actually kinda disappointed with this build after getting it all set up and trying to play with the lights using RGBFusion. Then I found Aurora, which was been working great (and it looks like you're the one who made the original RGBFusion390SetColor which is what got Aurora working in the first place) and now this extra customization is icing on the cake.

Cheerpipe committed:
Optimize communication latency and use latest Bridge

Don't require specific RGBFusion. Feel free to test your favorite version.
Testing bulk commands to reduce latency. It should show always 0 in Aurora device list.
Update MD5 tables to the latest Bridge and remove patched LedLib2.dll installation.

Cheerpipe commented:
> Is the latest version slowed down to work with the new RGBFusion? And is there a way to detect which version is installed, so the user doesn't need to set it?

Latest version should be RGBFusion version independant. 

> Not really sure how to measure that lol. If I had to guess, I'd say 25ish changes per second. Its fast enough that gradients are smooth unless the 2 colors are very different.

Nice. In some hardware configurations framerates are about 1 change per second.

> So those auto populate by what you've got? That would mean there are 11 more zones I couldn't find any lights attached to (beyond the ones I marked as ???? up there).

Areas are detected by RGBFusion. I just use this array. I also have non working areas :/.

Hope lastest version work fine because it has a lot of deep changes :(

skielbasa commented:
> Latest version should be RGBFusion version independant.

Geeze. you're on top of this.

> Nice. In some hardware configurations framerates are about 1 change per second.

Oh damn, yeah definitely not that slow.

> Areas are detected by RGBFusion. I just use this array. I also have non working areas :/

So weird, guessing they're either leftovers from some feature they're working on abandoned

rheniumtorn commented:
I've been testing this for a while and with the current build everything is working pretty great. I'm using an Aorus B450 Pro WIFI which has 3 LED zones along with 7 ARGB fans connected to 2 D_LED headers. The new ui for managing keybindings for different MB areas is a super nice addition along with the ability to set unused areas to be ignored (and doing this made the lighting refresh rate near-instant for me). 
![image](https://user-images.githubusercontent.com/66344014/83588957-8826c480-a520-11ea-88f8-fdcfb987b731.png)

Cheerpipe commented:
I know the code is not nice but still could be nice to have some review to start working on a final version or a mergeable versión because a think there will not be new features for now.

Frosthaven commented:
Clean code is always something to strive for, but considering I'm already keeping my RGB synced on strings and bubble gum I'm more than willing to take perfect code that still makes life easier :)

skielbasa commented:
Hate to keep doing this to you... plans ended early, so I got a chance to try your latest build tonight. It doesn't seem to be able to control RAM. The sticks just go red and stay there restarting the device and the whole app don't help.

That was build v0.8.0+859. The previous build (v0.8.0+856) works fine. I'm on RGBFusion ver B19.0919.1

rheniumtorn commented:
> Hate to keep doing this to you... plans ended early, so I got a chance to try your latest build tonight. It doesn't seem to be able to control RAM. The sticks just go red and stay there restarting the device and the whole app don't help.
> 
> That was build v0.8.0+859. The previous build (v0.8.0+856) works fine. I'm on RGBFusion ver B19.0919.1

Just curious, but was RGBFusion controlling your RAM just fine on it's own? I ask because I have Corsair Vengeance Pro RGB RAM and for whatever reason I've never been able to get RGBFusion to control it.

Cheerpipe commented:
> Hate to keep doing this to you... plans ended early, so I got a chance to try your latest build tonight. It doesn't seem to be able to control RAM. The sticks just go red and stay there restarting the device and the whole app don't help.
> 
> That was build v0.8.0+859. The previous build (v0.8.0+856) works fine. I'm on RGBFusion ver B19.0919.1

Check this:
![image](https://user-images.githubusercontent.com/972765/83596672-320f4c80-a533-11ea-8d78-0f64a021a4de.png)

Try emptying that textbox.

Cheerpipe committed:
Remove default ignoreled list to avoid consufion

skielbasa commented:
> Try emptying that textbox. I disabled RAM and VGAs by default to avoid framerate drop and allow users to use ram own software. Maybe i should let this blank by default.

That did the trick, didn't even think to see if that list wasn't empty. I see you've got a new build where it's empty by default. I understand why you originally had them ignored, but this is probably the best for UX unless there are a bunch of reports of performance issues.

Cheerpipe committed:
Merge remote-tracking branch 'antonpup/dev' into RGBFusion

diogotr7 committed:
Merge pull request #2078 from antonpup/dev

v0.8.0-rc3

Cheerpipe committed:
Merge tag 'v0.8.0rc3' into RGBFusion

Cheerpipe committed:
Merge remote-tracking branch 'antonpup/dev' into RGBFusion

Hukuma1 commented:
Hope it gets merged soon. Thanks again for this!

Cheerpipe committed:
Merge remote-tracking branch 'antonpup/dev' into RGBFusion

Cheerpipe committed:
Better logging and better start stop

More logging for debug
Better start stop control
Wait until RGBFusion HAL is loaded and pipe is listening to send first command at Aurora's start.
Restore last color if device restart.
Code cleanup.
Updated RGBFusion Bridge
More flexible RGBFusion Bridge installation procedure

Cheerpipe committed:
Small performance tweak

Avoid scan key array for commit key

Cheerpipe committed:
Custom args support

Used to load custom drivers

Cheerpipe committed:
Update DeviceMap on device init

And code clean up

Cheerpipe committed:
Merge branch 'dev' into RGBFusion

skielbasa commented:
I think there's something wrong with RGBFusionAuroraListener.exe in the latest build. Launching the latest build seems to have trouble initializing the bridge and if it does, the latency is 99ms and colors don't change. Rolled back to Aurora-v0.8.0+859-hloahclc (and the bridge included in that version) and it works again.

Cheerpipe commented:
Hi, thanks fo le testing. Pls upload your logs.

skielbasa commented:
[2020-06-21 17.56.38.log](https://github.com/antonpup/Aurora/files/4810364/2020-06-21.17.56.38.log)

No problem, least I can do when you go through the effort of making it. Log should be attached.

What sticks out to me is:

RGBFusion Device has encountered an error while sending the command 01-FF-00-00-00-00-00 to RGBFusion Bridge. Error: The operation has timed out.

Cheerpipe committed:
Try to get better logging and fix null object exception

Use updated RGBFusion Bridge

Cheerpipe commented:
> [2020-06-21 17.56.38.log](https://github.com/antonpup/Aurora/files/4810364/2020-06-21.17.56.38.log)
> 
> No problem, least I can do when you go through the effort of making it. Log should be attached.
> 
> What sticks out to me is:
> 
> RGBFusion Device has encountered an error while sending the command 01-FF-00-00-00-00-00 to RGBFusion Bridge. Error: The operation has timed out.

> [2020-06-21 17.56.38.log](https://github.com/antonpup/Aurora/files/4810364/2020-06-21.17.56.38.log)
> 
> No problem, least I can do when you go through the effort of making it. Log should be attached.
> 
> What sticks out to me is:
> 
> RGBFusion Device has encountered an error while sending the command 01-FF-00-00-00-00-00 to RGBFusion Bridge. Error: The operation has timed out.

Can you try the new build and:

 * Check if listener is running.
 * Check if files RGBFusionAuroraListener.exe and RGBFusionBridge.dll are in your RGBFusion folder.
 * If not running, open it manually (as admin).

Also, what RGBFusion version are you using? It is working fine for me at least with these versions:

19.0919.1
20.0430.1
B20.0529.1

Thanks in advance.

skielbasa commented:
[2020-06-21 19.34.50.log](https://github.com/antonpup/Aurora/files/4810487/2020-06-21.19.34.50.log)

RAM goes all red when launching Aurora which is normal during the initialization, but colors never start and on this build RGBFusion in  Aurora devices never even makes it to started (even manually clicking start) EDIT: restarted the app yet again, this time its listed as running, but at 100+ latency and no colors change.

New Log
[2020-06-21 19.41.23.log](https://github.com/antonpup/Aurora/files/4810525/2020-06-21.19.41.23.log)


 RGBFusionAuroraListener.exe is running and I've got the dll and exe in the RGBFusion folder. Using the version of RGBFusion linked in the OP which should mean I'm on 19.0919.1

Cheerpipe commented:
Wait, you reinstall RGBFusion 19.0919.1 and copy bridge and dll from where? (My english is not good enough sometimes).

Your last log told me it is working now, right?

Can you try this build? Is the latest Aurora but with older bridge. There must some broke with x570 in the latest bridge.


[Debug.zip](https://github.com/antonpup/Aurora/files/4810555/Debug.zip)

Cheerpipe committed:
Revert "Revert back to old bridge version"

This reverts commit dea4ec88e61a99d7a64ab3e5dc148c73a72ee481.

skielbasa commented:
Still not working, it launches, but the latency is 100+ and no colors change.

19.0919.1 from a link you posted in the initial post.

Bridge and dll installed on their own when launching Aurora. When it the latest build didn't work, I rolled bridge and dll back from the last version that did work. If I run the new build, it updates them back to the version that doesn't work.

skielbasa commented:
Debug build doesn't even launch :/

EDIT: more details. I get the UAC prompt to run as admin, but after accepting it, main Aurora process starts and immediately stops (watching from task manager).

Cheerpipe commented:
Very strange, i don't have any problems.

Can you give me some minutes in Discord? You i am Cheerpipe in Aurora Channel

skielbasa commented:
No problem. Discord DM incoming. Discord and github names are the same

Cheerpipe committed:
Fix shutdown and speedup startup

Cheerpipe committed:
Add some more logging

skielbasa commented:
latest build launches, initializes, and seems to start but stops on the first color of the pattern. Nothing meaningful in the logs (install and launch show as success).

EDIT: Had to restart my computer and figured I'd give this build another shot. It's working for me now. Guess I needed a restart for it?

StyFleer commented:
Could you do a full introduction to install it? I have a Gigabyte Aorus z390 Elite and I don't know how to install. I tryied a lot of things but no sucess. I don't know anything about programming

StyFleer commented:
When I restarted my system, it appears: 
![problema](https://user-images.githubusercontent.com/67871504/86538527-e1419980-becc-11ea-90df-3ac7a41e2daa.png)

skielbasa commented:
@StyFleer do you have RGBFusion installed?

Installation process should be something along the lines of:
1. Install RGBFusion from Gigabyte's site
2. Run RGBFusion at least once (not sure if this is still required)
3. Install Aurora from https://ci.appveyor.com/project/antonpup/aurora/builds/33655172/artifacts (you can grab either the installer or the zip. If using the installer, just run it; if using the zip unzip it were you want it).
4. Run Aurora (make sure RGBFusion isn't currently running as it will fight Aurora for control)
5. In Aurora settings go to devices and you should see an item for Gigabyte. Make sure it's enabled and running.
6. You can edit what keys are bound to what lighting zones by clicking options on the device. All zones are bound to ESC by default. Any changes here require relaunching Aurora to update. As of now zones are just numbered 1-however many. To figure out what number zone corresponds to what output on the board you'll need to experiment as its different for every model, that said further up this thread I have a table of zone mappings for my Aourus x570 Pro WIFI which should at least point you in the right direction.

To expand on the key binding bit at its heart Aurora is am effects engine /synchronizer for RGB peripherals, predominantly keyboards. Until custom device maps are added, extras like motherboards, fans, controllers, etc are currently mapped to keyboard keys and will light up based on that key's lighting. Eg if I set my RGBA header 1 to the Q  key, then whatever lighting Aurora applies to 1 will also apply to my devices plugged into RGBA header 1. As far as actually setting up lighting effects, the main Aurora site has pretty good tutorials and docs. https://aurora-rgb.github.io/Docs/basic-topics/getting-started/

StyFleer commented:
> @StyFleer do you have RGBFusion installed?
> 
> Installation process should be something along the lines of:
> 
> 1. Install RGBFusion from Gigabyte's site
> 2. Run RGBFusion at least once (not sure if this is still required)
> 3. Install Aurora from https://ci.appveyor.com/project/antonpup/aurora/builds/33655172/artifacts (you can grab either the installer or the zip. If using the installer, just run it; if using the zip unzip it were you want it).
> 4. Run Aurora (make sure RGBFusion isn't currently running as it will fight Aurora for control)
> 5. In Aurora settings go to devices and you should see an item for Gigabyte. Make sure it's enabled and running.
> 6. You can edit what keys are bound to what lighting zones by clicking options on the device. All zones are bound to ESC by default. Any changes here require relaunching Aurora to update. As of now zones are just numbered 1-however many. To figure out what number zone corresponds to what output on the board you'll need to experiment as its different for every model, that said further up this thread I have a table of zone mappings for my Aourus x570 Pro WIFI which should at least point you in the right direction.
> 
> To expand on the key binding bit at its heart Aurora is am effects engine /synchronizer for RGB peripherals, predominantly keyboards. Until custom device maps are added, extras like motherboards, fans, controllers, etc are currently mapped to keyboard keys and will light up based on that key's lighting. Eg if I set my RGBA header 1 to the Q key, then whatever lighting Aurora applies to 1 will also apply to my devices plugged into RGBA header 1. As far as actually setting up lighting effects, the main Aurora site has pretty good tutorials and docs. https://aurora-rgb.github.io/Docs/basic-topics/getting-started/

Thank you very much about the answer. The problem here is: in Aurora app don't appears Gigabyte on Devices e Wrappers, even in Device Manager.
There is a lot of brands, but don't have Gigabyte.
Do you know how could I do it?

StyFleer commented:
> @StyFleer do you have RGBFusion installed?
> 
> Installation process should be something along the lines of:
> 
> 1. Install RGBFusion from Gigabyte's site
> 2. Run RGBFusion at least once (not sure if this is still required)
> 3. Install Aurora from https://ci.appveyor.com/project/antonpup/aurora/builds/33655172/artifacts (you can grab either the installer or the zip. If using the installer, just run it; if using the zip unzip it were you want it).
> 4. Run Aurora (make sure RGBFusion isn't currently running as it will fight Aurora for control)
> 5. In Aurora settings go to devices and you should see an item for Gigabyte. Make sure it's enabled and running.
> 6. You can edit what keys are bound to what lighting zones by clicking options on the device. All zones are bound to ESC by default. Any changes here require relaunching Aurora to update. As of now zones are just numbered 1-however many. To figure out what number zone corresponds to what output on the board you'll need to experiment as its different for every model, that said further up this thread I have a table of zone mappings for my Aourus x570 Pro WIFI which should at least point you in the right direction.
> 
> To expand on the key binding bit at its heart Aurora is am effects engine /synchronizer for RGB peripherals, predominantly keyboards. Until custom device maps are added, extras like motherboards, fans, controllers, etc are currently mapped to keyboard keys and will light up based on that key's lighting. Eg if I set my RGBA header 1 to the Q key, then whatever lighting Aurora applies to 1 will also apply to my devices plugged into RGBA header 1. As far as actually setting up lighting effects, the main Aurora site has pretty good tutorials and docs. https://aurora-rgb.github.io/Docs/basic-topics/getting-started/

I sent for you a friendship invitation on discord. If you could accept me and help, I would aprecciate that.

Cheerpipe committed:
Merge remote-tracking branch 'antonpup/dev' into RGBFusion

Cheerpipe committed:
Listener auto reset

skielbasa commented:
Latest build is working fine

Cheerpipe committed:
Merge remote-tracking branch 'antonpup/dev' into RGBFusion

Cheerpipe committed:
Merge branch 'dev' into RGBFusion

Cheerpipe committed:
Fix build not copying bridge files and update bridge version

Hukuma1 commented:
@diogotr7 Can this please be merged as well? Looking for native RGB Fusion support :)

Cheerpipe commented:
> @diogotr7 Can this please be merged as well? Looking for native RGB Fusion support :)

I think it needs more testing to avoid people blaming over Aurora.

skielbasa commented:
The last few builds have worked great as my main install. What specific testing do you need? Or just more different hardware?

Cheerpipe commented:
Just more hardware and configurations

skielbasa commented:
Fair. Unfortunately, I can't help with that one :/

hentzarfati commented:
**My Build:**
* RGBFusion Version B19.0919 (also tried all other versions of B20.XX, they also works)
* GPU: Gigabyte AORUS GeForce® RTX 2080 SUPER™ 8G
* Motherboard: Gigabyte X570 Aorus Ultra
* Windows 10 Version 1909 build 18363.959
* RAM: Corsair Vengeance RGB PRO 16GB (2x8GB) DDR4 3200MHz C16

**Issues:** 
**RAM**
Cant use RGBFusion to control RAM LEDs (RAM Sticks are not synced, not smooth, and blinks - not stable).
but this is not a problem with this bridge as I had this problem even when using RGBFusion without any other software.
I managed to solve this problem by adding ram area (index 8) to ignore list in "RGBFusion Options" and use iCUE to manage RAM LEDs (I had to enable the option: "Enable full software control" in iCUE settings)
That's also increased LEDs framerate for other accessories that controlled by RGBFusion (Front FANs & Motherboard LEDs) via the adapter.

I also ignored indexes 1,2,5,6,7 as I don't have any LEDs attached to them (that didn't improve the framerate).

**GPU**
The only real issue that I have now and didn't have when using RGBFusion by itself is with the GPU LEDs.
They are not stable, not running smoothly, and also display colors that not configured in Aurora.

I have seen in the source code of the adapter that there is some special care for Aorus 2080 cards by adding this command-line argument: "--aorusvgadriver ", but that didn't fix the problem for me.

I attached zip file that contains a video that shows the problem with the GPU LEDs, you can see that Motherboard LEDs, RAM LEDs and Front Fans LEDs are working in sync and smooth.

I hope my post is helping you with the testing and also hope you can help with my problem :)

Thanks.

[VGA LEDs problem.zip](https://github.com/antonpup/Aurora/files/5030582/VGA.LEDs.problem.zip)

Cheerpipe commented:
> **My Build:**
> 
> * RGBFusion Version B19.0919 (also tried all other versions of B20.XX, they also works)
> * GPU: Gigabyte AORUS GeForce® RTX 2080 SUPER™ 8G
> * Motherboard: Gigabyte X570 Aorus Ultra
> * Windows 10 Version 1909 build 18363.959
> * RAM: Corsair Vengeance RGB PRO 16GB (2x8GB) DDR4 3200MHz C16
> 
> **Issues:**
> **RAM**
> Cant use RGBFusion to control RAM LEDs (RAM Sticks are not synced, not smooth, and blinks - not stable).
> but this is not a problem with this bridge as I had this problem even when using RGBFusion without any other software.
> I managed to solve this problem by adding ram area (index 8) to ignore list in "RGBFusion Options" and use iCUE to manage RAM LEDs (I had to enable the option: "Enable full software control" in iCUE settings)
> That's also increased LEDs framerate for other accessories that controlled by RGBFusion (Front FANs & Motherboard LEDs) via the adapter.
> 
> I also ignored indexes 1,2,5,6,7 as I don't have any LEDs attached to them (that didn't improve the framerate).
> 
> **GPU**
> The only real issue that I have now and didn't have when using RGBFusion by itself is with the GPU LEDs.
> They are not stable, not running smoothly, and also display colors that not configured in Aurora.
> 
> I have seen in the source code of the adapter that there is some special care for Aorus 2080 cards by adding this command-line argument: "--aorusvgadriver ", but that didn't fix the problem for me.
> 
> I attached zip file that contains a video that shows the problem with the GPU LEDs, you can see that Motherboard LEDs, RAM LEDs and Front Fans LEDs are working in sync and smooth.
> 
> I hope my post is helping you with the testing and also hope you can help with my problem :)
> 
> Thanks.
> 
> [VGA LEDs problem.zip](https://github.com/antonpup/Aurora/files/5030582/VGA.LEDs.problem.zip)

Hi, thanks for your report.

The RAM issues is a known issue. Most of ram uses SMBUS to control lights. This is slow and unstable. Ignoring RAM area is the best you can do.


For the VGA, pls dont use --aorusvgadriver, It writes directly into i2c bus of the card but is specifically for the Aorus 2080 GAMINC OC model.

Is RGBFusion closes? If it is opened after aurora can cause the effect you show in your video.

Bridge has some DLED code control areas over ID 9. How many areas does your setup have?

skielbasa commented:
No issues on the latest build, and thank you for either updating or merging a branch that updated the build number. Now the update nag is gone on startup!

skielbasa commented:
> They are not stable, not running smoothly, and also display colors that not configured in Aurora.

I've got the 2070 version of that card and had the same issue with any control software that's not RGBFusion (tried a few of the other open source projects before settling on Aurora), so I don't think that is Aurora or Fusion Bridge related. 

Do you happen to have Aorus Engine installed?

.
.
.

I actually covered over the logos with some custom cut vinyl... they're kinda ugly anyways + my GPU is vertical mounted (note: the fan lighting works perfect).

hentzarfati commented:
@skielbasa 
I don't have Aorus Engine installed and my GPU mounted horizontally

@Cheerpipe
1. "For the VGA, pls dont use --aorusvgadriver, It writes directly into i2c bus of the card but is specifically for the Aorus 2080 GAMINC OC model." 

Thank you, I've disabled --aorusvgadriver, but still got the same problem.

2. "Is RGBFusion closes? If it is opened after aurora can cause the effect you show in your video." 

RGBFusion is closed when using Aurora.

3. "Bridge has some DLED code control areas over ID 9. How many areas does your setup have?"

By looking inside  "Aurora -> Device Manager -> RGB Fusion (View Options)" I can set 32 areas (from area index 0 to 33. area index 7 is missing)

I am attaching another 2 videos I took now, after removing "--aorusvgadriver" from custom command arguments,
The videos show that wrong colors are being used (red/orange/yellow/green shouldn't be there) and LEDs continue to flick

[gpu wrongs colors and flicks.zip](https://github.com/antonpup/Aurora/files/5035766/gpu.wrongs.colors.and.flicks.zip)

skielbasa commented:
@hentzarfati If you check this post https://github.com/antonpup/Aurora/pull/2057#issuecomment-637821771, your card likely has all the same indexes as mine (ignore the key column that was just the key I assigned the index to for testing) which should give you better control over the other lights.

That said, the fan stop and side logos have always given me troubles with anything other than RGBFusion when not set to a static color. If I had to guess, the GPU probably defaults to its rainbow cycle while a new color is written to the card's memory, which is why color changing causes the flicker but static solids don't. Though I couldn't say why it would only affect the logos and not the fans if that were the case.

xFuty commented:
My Build:

- RGBFusion Version B20.0708.1 ( I need to use a newer version because B19 doesn't support my motherboard)
- GPU: Gigabyte Radeon RX 5700XT Gaming OC
- Motherboard: Gigabyte B550 Aorus Elite
- Windows 10 Version 2004 build 19041.388


Ever since I installed the B550 motherboard neither it nor the GPU respond to color changes in Aurora. When I used the GPU with an older motherboard and RGBFusion Version B19.0919 it was working as intended. In the Device Manager RGB Fusion shows up as Connected with a latency of 0 ms. It also shows multiple areas (index 0-8, except 7. It matches the number of areas it's supposed to have). If I try to use version B19.0919 it can't even connect. Other B19. versions I tried didn't work either.

skielbasa commented:
> Ever since I installed the B550 motherboard neither it nor the GPU respond to color changes in Aurora. When I used the GPU with an older motherboard and RGBFusion Version B19.0919 it was working as intended. In the Device Manager RGB Fusion shows up as Connected with a latency of 0 ms. 

Just double checking, you don't have RGBFusion running at the same time as Aurora do you? If so, make sure RGBFusion isn't running. It needs to be installed and ran once (to build the index), but after that it doesn't need be / shouldn't be launched when using Aurora. And does stopping/restarting the RGB Fusion entry in Aurora devices help?

Sorry for the late response.

xFuty commented:
I ran RGB Fusion once. And I made sure to check in Task-Manager that it isn't running before I start Aurora. Restarting the RGB Fusion entry doesn't help either.

Cheerpipe commented:
> My Build:
> 
> * RGBFusion Version B20.0708.1 ( I need to use a newer version because B19 doesn't support my motherboard)
> * GPU: Gigabyte Radeon RX 5700XT Gaming OC
> * Motherboard: Gigabyte B550 Aorus Elite
> * Windows 10 Version 2004 build 19041.388
> 
> Ever since I installed the B550 motherboard neither it nor the GPU respond to color changes in Aurora. When I used the GPU with an older motherboard and RGBFusion Version B19.0919 it was working as intended. In the Device Manager RGB Fusion shows up as Connected with a latency of 0 ms. It also shows multiple areas (index 0-8, except 7. It matches the number of areas it's supposed to have). If I try to use version B19.0919 it can't even connect. Other B19. versions I tried didn't work either.


Can you try with B20.0529.1? I also run into throbless with B20.0708.1.

I am in process of upgrading my mainboard to a Z490 Aorus Master. I will try to work on B20.0708.1 compatibility this weekend.

xFuty commented:
Using B20.0529.1 fixed the problem for me! All the lights change as they should. The only issue I have is a slight delay.

skielbasa commented:
That's usually all it takes. With that in mind, it might be helpful to post
your logs for the Dev, he might know what's happening. Debug tab >  show
logs folder.

On Wed, Aug 19, 2020, 5:17 PM xFuty <notifications@github.com> wrote:

> I ran RGB Fusion once. And I made sure to check in Task-Manager that it
> isn't running before I start Aurora. Restarting the RGB Fusion entry
> doesn't help either.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/antonpup/Aurora/pull/2057#issuecomment-676709886>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABG4SOMTAW7VR7UNAZJC723SBQ6NRANCNFSM4NPIPEZA>
> .
>

xFuty commented:
Sure, I attached the log below.

[2020-08-20 13.52.34.log](https://github.com/antonpup/Aurora/files/5102768/2020-08-20.13.52.34.log)

Hukuma1 commented:
I had to reinstall and noticed RGB Fusion got updated to: B20.0721.1

Doesn't seem to be working, but I want to confirm it's not something on my end. Can anyone confirm? Or also send a link to the last working one please?

skielbasa commented:
> I had to reinstall and noticed RGB Fusion got updated to: B20.0721.1
> 
> Doesn't seem to be working, but I want to confirm it's not something on my end. Can anyone confirm? Or also send a link to the last working one please?

Here's the version the dev originally recommended.

https://drive.google.com/file/d/1inUFhNrpah1WZdJQv07SwKYnVfJ6DakL/view

Cheerpipe committed:
Merge remote-tracking branch 'antonpup/dev' into RGBFusion

Cheerpipe committed:
Update to new interface

Thanks to https://github.com/diogotr7/Aurora/blob/RGBFusion/Project-Aurora/Project-Aurora/Devices/RGBFusion/RGBFusionDevice.cs

Cheerpipe committed:
Code clean up and better rgbfusion handling

Fix lights not changing of commit key was not used.
Kill RGBFusion process on device start
Relax autoreset time

Cheerpipe committed:
Adjust device timeout

Cheerpipe committed:
Removed submodule NZXTSharp

rheniumtorn commented:
My LEDs are updating even faster with the newest build; it seems practically perfect now. Whatever you did, great work!

ikrowni commented:
Awesome I cant wait to see this supported. This will also work for the x470 boards?

Cheerpipe commented:
Should. May you try and report back?

El vie., 16 de octubre de 2020 9:08 p. m., KROWN <notifications@github.com>
escribió:

> Awesome I cant wait to see this supported. This will also work for the
> x470 boards?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/antonpup/Aurora/pull/2057#issuecomment-710703910>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHNPXLFLKZ5RZXQW5AK4KLSLDN6JANCNFSM4NPIPEZA>
> .
>

ikrowni commented:
> Should. May you try and report back?

How do i install it to try? The installation portion says 

"This application does not need to be installed. You just have to copy RGBFusionCli.exe into the [Gigabyte's RGB Fusion Application] folder. Typically C:\Program Files (x86)\GIGABYTE\RGBFusion."

Sounds simple enough but when i download this

https://github.com/Cheerpipe/RGBFusionCli/archive/master.zip

i dont see a file named (RGBFusionCli.exe)

ikrowni commented:
> Should. May you try and report back?

What do i need to do after copying the RGBFusionCli.exe to the proper folder?

Cheerpipe commented:
> > Should. May you try and report back?
> 
> What do i need to do after copying the RGBFusionCli.exe to the proper folder?

Just try this build

https://ci.appveyor.com/project/antonpup/aurora/builds/35703540/artifacts

No need of copy anything.

ikrowni commented:
Just finished trying it and sadly its not working for me.

I uninstalled RGBFusion and installed version B19.0919.1 as mentioned above.

I restarted my computer opened rgbfusion. Closed out rgb fusion completely then started aurora. Inside aurora i see the RGBFusion under device manager. Its options say it is bound with the esc key.  But when i try to get some lighting going its not working. Now im pretty new to aurora but everything else seems to be working fine keyboard, mouse, wled light strips, corsair ram.

I have 9 case fans, a water pump, and aorus 2080 GPU hooked to my aorus x470 mobo that still need to be synced.

MatrixDJ96 commented:
Thanks for your amazing work @Cheerpipe!

I'm using AORUS Z390 PRO WIFI and it works with Aurora
RGBFusion bridge is also working with my HyperX Predator RGB!

I suggest you work on smoothness because is the only thing missing to make RGBFusionBrigde perfect

Hukuma1 commented:
> > I had to reinstall and noticed RGB Fusion got updated to: B20.0721.1
> > Doesn't seem to be working, but I want to confirm it's not something on my end. Can anyone confirm? Or also send a link to the last working one please?
> 
> Here's the version the dev originally recommended.
> 
> https://drive.google.com/file/d/1inUFhNrpah1WZdJQv07SwKYnVfJ6DakL/view

Thanks! This worked. I was having issues with my RAM all of a sudden not showing up.

Apparently if you ever installed RGB Fusion and any additional programs like Trident-Z RGB it could conflict and remain un-working. Fortunately, the solution is simple. GSkill made a cleaner batch script that fully removes any remnants of old drivers: https://www.gskill.us/forum/forum/product-discussion/ddr4/trident-z-rgb/163741-trident-z-lighting-control-driver-conflict-clean-uninstall-guide

Massive thanks to @Cheerpipe for getting RGB Fusion support into Aurora!

AFaustini commented:
![image](https://user-images.githubusercontent.com/32999190/98270884-1985e100-1f6e-11eb-8763-277611e8db5b.png)
How can i find information about where each "area index" is?

carlonex commented:
Cheerpipe you are a genius, your integration with rgb fusion worked perfectly for me both memory and motherboard, i use a b450pro aorus i wanted to know if i can integrate with the cooler master mp750 xl.please i need help it is the only device that does not synchronize

Cheerpipe commented:
Hi @carlonex 

I not working in this device anymore because I moved to Artemis mainly because it support more than just keyboards and mouses and it is easy to develope plugins and device providers.

I recommend you to give a try to Artemis+OoenRGB. It could be hard to start with but it the end you can do almost anything.

https://artemis-rgb.com/

https://discord.gg/UsW5RAYt

ikrowni commented:
> I recommend you to give a try to Artemis+OoenRGB.

Just checked out Artemis but while searching for videos of it I saw another called (Signal RGB) do you have any thoughts on it vs Artemis?

Cheerpipe commented:
> > I recommend you to give a try to Artemis+OoenRGB.
> 
> Just checked out Artemis but while searching for videos of it I saw another called (Signal RGB) do you have any thoughts on it vs Artemis?

I didn't tried signal RGB mainly because it is not free nor opensource still looks like a good propietary software alternative

Hukuma1 commented:
Sad to hear that RGBFusionCli is no more, Cheerpipe! Thank you for your hard work.

As far as OpenRGB there is an open issue with RGBFusion unfortunately:

https://gitlab.com/CalcProgrammer1/OpenRGB/-/issues/1243

Hope there will be a plugin for Artemis?

mirh commented:
That's for newer boards. 
Something older like OP's should already be more or less supported (help is always welcome tho)
https://gitlab.com/CalcProgrammer1/OpenRGB/-/issues/118
